### PR TITLE
Add documentation re: Metalsmith collections

### PR DESCRIPTION
**Pull request by ghost**
Adds a short DevOps-focused version of how to add collections in Metalsmith. 

Related to department-of-veterans-affairs/vets.gov-team#2481

### CODE REVIEW

**Comment by tiffanybbrown-va** [comment]
This is one of those weird intersections between content and DevOps. I would like someone DevOps-y to read through it, just in case it isn't clear.

**Comment by ayaleloehr** [comment]
cc @jkassemi @patrickvinograd @CyberKoz from devops, per comment above.

**Comment by raquelromano-gov** [comment]
Just noticed that https://github.com/department-of-veterans-affairs/vets-website/blob/master/docs/WhyIsMyBuildBreaking.md links to HowURLsAreCreated.md, so may want to update or replace that link when the file goes away

File: docs/HowURLsAreCreated.md [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-  <li class="parent">&lt;a href="/vre/">VR&amp;E&lt;/a></li>
-  <li class="parent">&lt;a href="/vre/family/">Family&lt;/a>&lt;/li>
-  <li class="active">Family Members&lt;/li>
-</ul>
-```
-And read _VR&E &gt; Family &gt; Family Members_. 
-
-## IMPORTANT
-
-_**Make sure that every collection sub-directory also has a Markdown file that shares its name.**_ Take the the `_employment/job-seekers/` directory as an example. It also has an `_employment/job-seekers.md` file. Each Markdown file in `_employment/job-seekers/` compiles to  `employment/jobs-seekers/name_of_markdown_file/index.html`, while `_employment/job-seekers.md` itself compiles to `employment/job-seekers/index.html`.
```
Comment (raquelromano-gov) [inline docs/HowURLsAreCreated.md]: This little bit seems useful---is there some other file we can add this to?

File: docs/HowtoCreateCollectionsOrRelatedLinksInMetalsmith.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# Adding a collection of related links
+_Added: 2017-04-27_
+
+Our current publishing system, Metalsmith, provides a way to group related content using what it calls _collections_. Using collections requires the `metalsmith-collections` plugin ([documentation](https://github.com/segmentio/metalsmith-collections)).
+
+Content grouped within a collection is displayed in the sidebar navigation. Viewing `content/components/navigation-sidebar.html` may clarify some of what's in this document.
```
Comment (melwoodard) [inline docs/HowtoCreateCollectionsOrRelatedLinksInMetalsmith.md]: Should the fact that they're also used to define the previous/next navigation be mentioned here too?

File: docs/HowtoCreateCollectionsOrRelatedLinksInMetalsmith.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        name: 'Title for collection two'
+      }
+    }
+  }));
+```
+
+### What each field means:
+
+- `collectionName`: should be a camel-cased or snake cased string that summarizes the name of the related documents.
+- `pattern`: should be a directory a relative to the site root (without a leading slash), and end with `*.md`.
```
Comment (melwoodard) [inline docs/HowtoCreateCollectionsOrRelatedLinksInMetalsmith.md]: It'd helpful for non-technical folks to explain that the `/*.md` means the collection will include all of the markdown files within that folder.

**Review by melwoodard — COMMENTED** [review]
@tiffanybbrown-va I added a couple comments based on our walkthrough today.

File: docs/HowURLsAreCreated.md (hunk 1 above)
Comment (tiffanybbrown-va) [inline docs/HowURLsAreCreated.md]: I'm not sure whether this is still relevant for Metalsmith. It's how our files are currently organized. But I don't think this is actually required (which is why I deleted the entire thing). 

@rroueche do you think it's helpful to retain this information somewhere?

**Comment by melwoodard** [comment]
@jkassemi @patrickvinograd @CyberKoz can you take a look at this documentation and give any feedback for Tiffany before EOW?

File: docs/HowURLsAreCreated.md (hunk 1 above)
Comment (raquelromano-gov) [inline docs/HowURLsAreCreated.md]: Defer to @rroueche to make the call!

**Review by jkassemi — APPROVED** [review]
Documentation is thorough and straightforward. I'm up to assist the content team in adding new collections when necessary. 👍 

@rroueche - any feedback on the "How Are URLs Created" section?

**Comment by rroueche** [comment]
@tiffanybbrown-va and @jkassemi is this part still true?
image

I thought breadcrumbs weren't generated by file path name anymore, but by display title (or page title as the fallback).

**Comment by tiffanybbrown-va** [comment]
No, that's not true anymore. Most of that document is out of date, which is why I'm proposing we delete it. However this part is still mostly-accurate. We're trying to determine whether to restore that portion of it.

> -## IMPORTANT
> -_**Make sure that every collection sub-directory also has a Markdown file that shares its name.**_ Take the the `_employment/job-seekers/` directory as an example. It also has an `_employment/job-seekers.md` file. Each Markdown file in `_employment/job-seekers/` compiles to  `employment/jobs-seekers/name_of_markdown_file/index.html`, while `_employment/job-seekers.md` itself compiles to `employment/job-seekers/index.html`.

**Comment by rroueche** [comment]
Ah ok, sorry i misunderstood. This seems good from my perspective, though I think we'll definitely take you up on your offer to help @jkassemi when we need to do this :)

**Comment by tiffanybbrown-va** [comment]
I'll restore the _-Make sure that every collection sub-directory also has a Markdown file that shares its name._ portion of HowURLsAreCreated.md. We can always delete it later.

**Comment by tiffanybbrown-va** [comment]
I'll merge at EOD if no one objects to these updates.

**Pull request merged**